Automattic/wp-calypso
# Help Center: remove unused file and fix article link

#### Changes proposed in this Pull Request

* Removes a left-over file from [this PR](https://github.com/Automattic/wp-calypso/pull/63502). 
* Fixes the ID and link for user roles article.
* Fixes tiny typo.

#### Testing instructions

1. Open the HC. 
2. Make sure "Inviting Contributors, Followers, and Viewers" works.
3. Make sure there is no regression in the inline-help due to the deleted file.
